inference/core/roboflow_api.py

```diff
@@ -0,0 +1,198 @@
+                logger.error(f"Could not connect to Roboflow API. Error: {error}")
+                raise RoboflowAPIConnectionError(
+                    "Could not connect to Roboflow API."
+                ) from error
+            except requests.exceptions.HTTPError as error:
+                logger.error(
+                    f"HTTP error encountered while requesting Roboflow API response: {error}"
+                )
+                user_handler_override = (
+                    http_errors_handlers if http_errors_handlers is not None else {}
+                )
+                status_code = error.response.status_code
+                default_handler = DEFAULT_ERROR_HANDLERS.get(status_code)
+                error_handler = user_handler_override.get(status_code, default_handler)
+                if error_handler is not None:
+                    error_handler(error)
+                raise RoboflowAPIUnsuccessfulRequestError(
+                    f"Unsuccessful request to Roboflow API with response code: {status_code}"
+                ) from error
+            except requests.exceptions.InvalidJSONError as error:
+                logger.error(
+                    f"Could not decode JSON response from Roboflow API. Error: {error}."
+                )
+                raise MalformedRoboflowAPIResponseError(
+                    "Could not decode JSON response from Roboflow API."
+                ) from error
+
+        return wrapper
+
+    return decorator
+
+
+@wrap_roboflow_api_errors()
+def get_roboflow_workspace(api_key: str) -> WorkspaceID:
+    api_url = wrap_url("/".join([API_BASE_URL, f"?api_key={api_key}"]))
+    api_key_info = requests.get(api_url)
+    api_key_info.raise_for_status()
+    workspace_id = api_key_info.json().get("workspace")
+    if workspace_id is None:
+        raise WorkspaceLoadError(f"Empty workspace encountered, check your API key.")
+    return workspace_id
+
+
+@wrap_roboflow_api_errors()
+def get_roboflow_dataset_type(
+    api_key: str, workspace_id: WorkspaceID, dataset_id: DatasetID
+) -> TaskType:
+    api_url = wrap_url(
+        "/".join(
+            [API_BASE_URL, workspace_id, dataset_id, f"?api_key={api_key}&nocache=true"]
+        )
+    )
+    dataset_info = requests.get(api_url)
+    dataset_info.raise_for_status()
+    project_task_type = dataset_info.json().get("project", {})
+    if "type" not in project_task_type:
+        logger.warning(
+            f"Project task type not defined for workspace={workspace_id} and dataset={dataset_id}, defaulting "
+            f"to object-detection."
+        )
+    return project_task_type.get("type", "object-detection")
+
+
+@wrap_roboflow_api_errors(
+    http_errors_handlers={
+        500: lambda e: raise_from_lambda(
+            e,
+            RoboflowAPINotNotFoundError,
+            "Could not find requested Roboflow resource. Check for correctness of workspace name, dataset or version",
```

Comment:
Change "Check for correctness of workspace name, dataset or version." to "Check that the provided dataset and version are correct, and check that the provided Roboflow API key has the correct permissions."

Reply:
👍